metric_learn/_util.py

```diff
@@ -12,39 +15,314 @@ def vector_norm(X):
     return np.linalg.norm(X, axis=1)
 
 
-def check_tuples(tuples):
-  """Check that the input is a valid 3D array representing a dataset of tuples.
-
-  Equivalent of `check_array` in scikit-learn.
+def check_input(input_data, y=None, preprocessor=None,
+                type_of_inputs='classic', tuple_size=None, accept_sparse=False,
+                dtype="numeric", order=None,

```

Comment:
Super nitpick: use single quotes for `dtype='numeric'` to keep consistency with the rest of the file.

Reply:
I agree, thanks 
Done